src/main/resources/META-INF/plugin.xml

```diff
@@ -49,6 +49,12 @@
                 <override-text place="GitMacheteContextMenu"/>
             </action>
 
+            <action id="GitMachete.SquashSelectedAction"
+                    class="com.virtuslab.gitmachete.frontend.actions.contextmenu.SquashSelectedAction"
+                    icon="MacheteIcons.SQUASH">
+                <override-text place="GitMacheteContextMenu"/>
```

Comment:
Huh know what?... I'd be even inclined to put Squash in **toolbar** menu, instead of Merge action (which has been added coz users requested... but actually should be discouraged in general: https://github.com/VirtusLab/git-machete-intellij-plugin/issues/1003). WDYT?

Reply:
sounds good